CMakeLists.txt

```diff
@@ -247,26 +246,23 @@ if(ENABLE_TESTCOVERAGE)
 endif()
 
 if(USE_MKLDNN)
-  include(cmake/DownloadMKLML.cmake)
   # CPU architecture (e.g., C5) can't run on another architecture (e.g., g3).
-  if(NOT MSVC)
-    set(ARCH_OPT_FLAGS "-mtune=generic")
-  else()
+  if(MSVC)
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /EHsc")
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /EHsc /Gy")
   endif()
 
-  set(WITH_TEST OFF CACHE INTERNAL "" FORCE)
-  set(WITH_EXAMPLE OFF CACHE INTERNAL "" FORCE)
-  set(ARCH_OPT_FLAGS "" CACHE INTERNAL "" FORCE)
+  set(MKLDNN_BUILD_TESTS OFF CACHE INTERNAL "" FORCE)
+  set(MKLDNN_BUILD_EXAMPLES OFF CACHE INTERNAL "" FORCE)
+  set(MKLDNN_ARCH_OPT_FLAGS "" CACHE INTERNAL "" FORCE)
+  set(MKLDNN_USE_MKL NONE CACHE INTERNAL "" FORCE)
+  set(MKLDNN_ENABLE_JIT_PROFILING OFF CACHE INTERNAL "" FORCE)
 
   add_subdirectory(3rdparty/mkldnn)
 
   include_directories(3rdparty/mkldnn/include)
   include_directories(${PROJECT_BINARY_DIR}/3rdparty/mkldnn/include)
-  add_definitions(-DUSE_MKL=1)
-  add_definitions(-DCUB_MKL=1)
-  add_definitions(-DMXNET_USE_MKLDNN=1)
+  add_definitions(-DMXNET_USE_MKLDNN=100)
```

Comment:
Could you elabroate where the 100 comes from?

Reply:
I simply take it as v1.0.0 of MKL-DNN. But actually this PR upgrades MKL-DNN dependency to the 1.0.1 patch release.
Anyway, by changing this, I want to skip the compilation of those code wrapped with `#if MXNET_USE_MKLDNN == 1` in the current code base as they are not compatible with the new MKL-DNN. Once a piece of the code (eg. the implementation of an operator) is changed to MKL-DNN 1.0 compatible, the PR author can change the check to `#if MXNET_USE_MKLDNN == 100` to make the new code can compile and link with the new MKL-DNN. That will happen incrementally.
Hope I make it clear for you.

Reply:
After all the integration is done, if we feel not comfortable with this `MXNET_USE_MKLDNN=100`, we can change it back to `MXNET_USE_MKLDNN=1` by search and replace.

Reply:
It seems a bit unintuitive it you're asking me. Especially because the term "USE" sounds like a Boolean flag. But what you're actually looking for here is a version check, right?

Do you have a timeline until everything has been migrated and what's the plan here exactly? It seems like his PR leaves the master branch in a kinda unusable state. So may I propose to instead have a separate branch to which you make all the changes and then merge that to master once everything has been migrated to 1.0?

Reply:
Oh nevermind, this is already a separate branch. It didn't show that on mobile, sorry. Go ahead :)